spec/elastic_apm/central_config_spec.rb

```diff
@@ -0,0 +1,176 @@
+# frozen_string_literal: true
+
+module ElasticAPM
+  RSpec.describe CentralConfig do
+    let(:config) { Config.new }
+    subject { described_class.new(config) }
+
+    describe '#start' do
+      it 'polls for config' do
+        req_stub = stub_response(transaction_sample_rate: '0.5')
+        subject.start
+        sleep 0.1
+        expect(req_stub).to have_been_requested
+      end
+
+      context 'when disabled' do
+        let(:config) { Config.new(central_config: false) }
+
+        it 'does nothing' do
+          req_stub = stub_response(transaction_sample_rate: '0.5')
+          subject.start
+          sleep 0.1
+          expect(req_stub).to_not have_been_requested
+        end
+      end
+    end
+
+    describe '#fetch_and_apply_config' do
+      it 'queries APM Server and applies config' do
+        req_stub = stub_response(transaction_sample_rate: '0.5')
+        expect(config.logger).to receive(:info)
+
+        subject.fetch_and_apply_config
+        sleep 0.1
+
+        expect(req_stub).to have_been_requested
+
+        expect(config.transaction_sample_rate).to eq(0.5)
+      end
+
+      it 'reverts config if later changed' do
+        stub_response(transaction_sample_rate: '0.5')
+
+        subject.fetch_and_apply_config
+        sleep 0.1
+
+        stub_response({})
+
+        subject.fetch_and_apply_config
+        sleep 0.1
+
+        expect(config.transaction_sample_rate).to eq(1.0)
+      end
+
+      it 'reverts config if later 404' do
+        stub_response(transaction_sample_rate: '0.5')
+
+        subject.fetch_and_apply_config
+        sleep 0.1
+
+        stub_response('Not found', status: 404)
+
+        subject.fetch_and_apply_config
+        sleep 0.1
+
+        expect(config.transaction_sample_rate).to eq(1.0)
+      end
+
+      context 'when server responds 200 and cache-control' do
+        it 'schedules a new poll' do
+          stub_response(
+            {},
+            headers: { 'Cache-Control': 'must-revalidate, max-age=123' }
+          )
+
+          subject.fetch_and_apply_config
+          sleep 0.1
+
+          expect(subject.task).to be_pending
+          expect(subject.task.initial_delay).to eq 123
+        end
+      end
+
+      context 'when server responds 304' do
```

Comment:
probably a more realistic test would be 200 followed by 304 - then you can also check that the 304 response doesn't cause the config to be reverted

Reply:
That's sort of what these test: https://github.com/mikker/apm-agent-ruby/blob/906b3b62616bfe272b1e8e043530ea23f6cdcb98/spec/elastic_apm/central_config_spec.rb#L157-L168

Reply:
I modified the 304 test for this